3.0 deprecations and bc (#664)
ezsystems/developer-documentation: docs/releases/ez_platform_v3.0_deprecations.md

> # eZ Platform v3.0 deprecations and BC breaks

Reviewer: ```suggestion
# eZ Platform v3.0 deprecations and backwards compatibility breaks
```

This page lists backwards compatibility breaks and deprecations introduced in eZ Platform v3.0.

## Backwards compatibility breaks

### Symfony 4

v3.0 now uses Symfony 4 instead of Symfony 3.
Refer to [Symfony changelog](https://github.com/symfony/symfony/blob/master/CHANGELOG-4.0.md)
and [Symfony upgrade guides](https://github.com/symfony/symfony/blob/master/UPGRADE-4.0.md)
to learn about all changes it entails.

> See [v3.0 project update](ez_platform_v3.0_project_update.md) for the steps you need to take to update your project to Symfony 4.

Reviewer: We can also add a link to [Requirements](../getting_started/requirements.md) it would fit nicely in this context.

### Field Types

> Tags used to register Field Type features in the dependency injection container have changed:

Reviewer: ```suggestion
Tags used to register Field Type features in the dependency injection container are renamed:
```
This part should be in Deprecations not Backwards compatibility breaks

> ```

Reviewer: ```suggestion
|Former name|New name|
|-----------|--------|
|`ezpublish.fieldType`|`ezplatform.field_type`|
|`ezpublish.fieldType.indexable`|`ezplatform.field_type.indexable`|
|`ezpublish.storageEngine.legacy.converter`|`ezplatform.field_type.legacy_storage.converter`|
|`ezpublish.fieldType.parameterProvider`|`ezplatform.field_type.parameter_provider`|
|`ezpublish_rest.field_type_processor`|`ezplatform.field_type.rest.processor`|
|`ez.fieldFormMapper.value`|`ezplatform.field_type.form_mapper.value`|
|`ez.fieldFormMapper.definition`|`ezplatform.field_type.form_mapper.definition`|
```

ezpublish.fieldType -> ezplatform.field_type
ezpublish.fieldType.indexable -> ezplatform.field_type.indexable
ezpublish.storageEngine.legacy.converter -> ezplatform.field_type.legacy_storage.converter
ezpublish.fieldType.parameterProvider -> ezplatform.field_type.parameter_provider
ezpublish_rest.field_type_processor -> ezplatform.field_type.rest.processor
ez.fieldFormMapper.value -> ezplatform.field_type.form_mapper.value
ez.fieldFormMapper.definition -> ezplatform.field_type.form_mapper.definition
```

> 

Reviewer: The 2 sentences below are BC breaks

`\eZ\Publish\Core\FieldType\FieldType::getName`, which had been deprecated, is in use again.

> The deprecated `eZ\Publish\Core\FieldType\RichText` namespace has been removed.

Reviewer: ```suggestion
The deprecated `eZ\Publish\Core\FieldType\RichText` namespace and `eZ\Publish\SPI\FieldType\Nameable` interface have been removed.
```

### Twig helper names

> Selected Twig helper names have been changed.

Reviewer: ```suggestion
Selected Twig helpers names have been changed.
```

|`ezpublish`|`ezplatform`|

##### Filters renamed:

|Former name|New name|
|-----------|--------|
|`richtext_to_html5`|`ez_richtext_to_html5`|
|`richtext_to_html5_edit`|`ez_richtext_to_html5_edit`|

### JavaScript event names and code cleanup

Selected event names have been changed.

#### In Admin UI (`ezplatform-admin-ui`):

|Former name|New name|
|-----------|--------|
|`invalidFileSize`|`ez-invalid-file-size`|
|`addressNotFound`|`ez-address-not-found`|
|`cancelErrors`|`ez-cancel-errors`|

#### In Page Builder (`ezplatform-page-builder`):

|Former name|New name|
|-----------|--------|
|`openUdw`|`ez-open-udw`|
|`openAirtimePopup`|`ez-open-airtime-popup`|
|`postUpdateBlocksPreview`|`ez-post-update-blocks-preview`|
|`pbIframeLoaded`|`ez-page-builder-iframe-loaded`|
|`pbHideTools`|`ez-page-builder-hide-tools`|

Additionally, the listener for `pbPreviewReloaded` has been removed.

#### In Form Builder (`ezplatform-form-builder`):

|Former name|New name|
|-----------|--------|
|`openUdw`|`ez-open-udw`|
|`updateFieldName`|`ez-update-field-name`|
|`fbFormBuilderLoaded`|`ez-form-builder-loaded`|
|`fbFormBuilderUnloaded`|`ez-form-builder-unloaded`

### REST server

> Removal of REST code from Kernel to a separate package results in the following change:

Reviewer: ```suggestion
Transfer of REST code from Kernel to a separate package results in the following change:
```

`eZ\Publish\Core\REST` and `eZ\Publish\Core\REST\Common\` namespaces have been replaced by `EzSystems\EzPlatformRest`.

REST client has been dropped.

### HTTP cache bundle

HTTP cache bundle now uses FOS Cache Bundle v2. This entails:

- `EzSystems\PlatformHttpCacheBundle\Proxy\TagAwareStore` has been removed
- `EzSystems\PlatformHttpCacheBundle\Handler\TagHandler` has been changed so that the tag is now provided as an option in `header_formatter`
- `tagResponse()` from `tagHandler` has been replaced by `tagSymfonyResponse()`
- deprecated `EzSystems\PlatformHttpCacheBundle\Handler\TagHandlerInterface` has been removed
- `EzSystems\PlatformHttpCacheBundle\PurgeClient\PurgeClientInterface` now only accept an array as argument in the `purge()` method, instead of an int.

### Deprecated Field Types

Deprecated `ezprice` and `ezpage` Field Types have been removed.

### Elastic Search

Elastic Search support has been dropped.

### Assetic support

Assetic support has been dropped.

### Miscellaneous

- Deprecated `SubtreeQuery` class has been removed. In v3.0 it was replaced by `\EzSystems\EzPlatformAdminUi\QueryType\SubtreeQueryType`.

## Deprecations

### Template parameter names

> Using the SiteAccess-aware `pagelayout` setting is deprecated, use `page_layout` instead.

Reviewer: ```suggestion
The SiteAccess-aware `pagelayout` setting is deprecated in favor of `page_layout`.
```

> View parameter `pagelayout` set by `pagelayout` setting is deprecated, use `page_layout` instead in your Twig templates.

Reviewer: ```suggestion
View parameter `pagelayout` set by `pagelayout` setting is deprecated  in favor of  `page_layout`.
```

### Template organization

The following templates used in the Back Office have been renamed:

> ??? note "Renamed templates"

Reviewer: Maybe table to keep it consistent?
